### PR TITLE
Pin libnetcdf

### PR DESCRIPTION
**Pull request by ocefpaf**
Should fix https://github.com/SciTools/conda-recipes-scitools/pull/149#issuecomment-210053682

Ping @pp-mo

### CODE REVIEW

File: netcdf-fortran/build.sh [hunk 1]
```diff
@@ -8,5 +8,7 @@ fi
 CPPFLAGS=-I$PREFIX/include LDFLAGS=-L$PREFIX/lib ./configure --prefix=$PREFIX
 
 make
-make check
+# By-passing the tests for now b/c compiling against an libnetcdf 4.2.1.1 causes
+# 2 failues: `ftst_rengrps` and `f90tst_rengrps`
```
Comment (ocefpaf) [inline netcdf-fortran/build.sh]: Hopefully this won't be a problem later on.

**Comment by ocefpaf** [comment]
I used all my tricks here but I cannot get `ESMF` to work on OS X!

See https://travis-ci.org/SciTools/conda-recipes-scitools/builds/124038681#L6431

File: .travis.yml [hunk 2]
```diff
@@ -25,12 +25,13 @@ addons:
             - check
 
 before_install:
+    - brew remove --force $(brew list)
```
Comment (ocefpaf) [inline .travis.yml]: I would not merge this before some careful consideration.

File: esmf/meta.yaml [hunk 3]
```diff
@@ -7,14 +7,18 @@ source:
...
 
 requirements:
     build:
+        - gcc
         - netcdf-fortran
+        - libnetcdf 4.2.1.1
     run:
+        - libgcc
         - netcdf-fortran
+        - libnetcdf 4.2.1.1
```
Comment (ocefpaf) [inline esmf/meta.yaml]: For some weird reason conda still tries to update `libnetcdf` even though `netcdf-fortran` has both `build` and `run` dependencies set to `libnetcdf 4.2.1.1`.

**Comment by ocefpaf** [comment]
@pp-mo You need only Linux for the testing, right? I guess I should just skip the build on OS X then and you'll have what you need. Sounds good?

**Comment by pp-mo** [comment]
> @ocefpaf You need only Linux for the testing, right?

I thought we did do automated testing against OSX, but I've just been informed that we don't !
So it is ok as long as ESMPy remains an _optional_ dependency.

**Comment by ocefpaf** [comment]
> I thought we did do automated testing against OSX, but I've just been informed that we don't !
So it is ok as long as ESMPy remains an _optional_ dependency.

Cool! Hopefully we will be able to use conda-forge version in the future.

@pelson if you are OK with this can you merge it?

**Comment by pp-mo** [comment]
Great @ocefpaf 
- I now have something working with all Iris tests !
I'll fix up our bridging routine + put up a PR to include it, this time with the testing in Travis.

**Comment by ocefpaf** [comment]
BTW I believe the Travis config will be OK with conda-forge packages. The change here was just for your  particular setup, no?

**Pull request merged**